### ISSUE TITLE
Retrieve batcher and job name from a JobTicket

### ISSUE THREAD

**Issue by kcoleman-marklogic** [marklogic/java-client-api#648]
At the moment you can retrieve a ticket from a `Batcher`, but you cannot retrieve a batcher from a ticket. That means you have to hang on to the batcher instead of to the ticket. 

Similarly, you can retrieve the ticket UUID from the ticket, but you cannot retrieve user-configurable job name that you can set using `Batcher.withJobName()`. Of course, I can get it from the batcher, but that again implies it is the Batcher and not the ticket I should hold on to. 

TL;DR Perhaps we should add the following to the `JobTicket` interface:

* `JobTicket.getBatcher()`
* `JobTicket.getJobName()`

**Comment by vivekmuniyandi** [comment]
JobTicket.getJobName() looks redundant once we have JobTicket.getBatcher() in place. As you said, once we have the batcher through JobTicket.getBatcher() we can get the name through Batcher.getJobName(). Spoke with Sam and he too felt the same. Let me know if we need to still have the getJobName() in the API. For now, I am going ahead with only the getBatcher(). Thanks.

**Comment by vivekmuniyandi** [comment]
```
Batcher JobTicket.getBatcher()
```
 This would get the corresponding Batcher instance from the JobTicket instance

**Comment by georgeajit** [comment]
QBFailover tests use this method.
